src/Game.cpp

```diff
@@ -31,6 +31,164 @@
+        return;
+    }
+
+    *esi = '\0';
+    glo_8600EC = (char**)MC2_MALLOC(4);
+
+    uint32_t length = strlen(args);
+    *glo_8600EC = MC2_STRDUP(args);
+
+    *esi = ' ';
+    esi++;
+
+    if (*esi == '\0') {
+        glo_8600F8.loc_611DC0();
+        return;
+    }
+
+    for (char *cur = esi; *cur != '\0'; ++cur) {
+        if (*cur != '-')
+            continue;
+
+        glo_8600F0++;
+
+        char *nextSeperator = cur;
+        while (*nextSeperator != '\0') {
+            if (*nextSeperator == ' ')
+                break;
+            if (*nextSeperator == '=')
+                break;
+            nextSeperator++;
+        }
+        char *key = cur + 1;
+        char *value_str = cur;
+        if (*nextSeperator != '\0') {
+            *nextSeperator = '\0';
+            value_str = nextSeperator + 1;
+        }
+        unk_612150 *value = glo_8600F8.sub_6124A0(key);
+
+        if (value != nullptr) {
+            cur = value_str;
+            continue;
+        }
+
+        value = (unk_612150*)MC2_MALLOC(sizeof(unk_612150));
+
+        if (value != nullptr) {
+            value->count = 0;
+            value->args = nullptr;
+        }
+        if (value == nullptr) {
+            // The original code forces an access violation instead
+            mc2_log_fatal("MC2 MALLOC Failed");
+        }
+
+        nextSeperator = value_str;
+        if (*nextSeperator != '\0') {
+            while (*nextSeperator != '-') {
+                value->count++;
+                if (*nextSeperator == 0)
+                    break;
+
+                do {
+                    if (*nextSeperator == '=')
+                        break;
+
+                    if (*nextSeperator == ' ')
+                        break;
+
+                } while (*nextSeperator++ != '\0');
+
+                if (*nextSeperator == '\0')
+                    break;
+
+                do {
+                    if (*nextSeperator == '=')
+                        break;
+
+                    if (*nextSeperator != ' ')
+                        break;
+
+                } while (*nextSeperator++ != '\0');
+            }
+        }
+        
+        cur = nextSeperator - 1;
+
+        glo_8600F8.sub_612150(key, value);
+
+        value->args = (char**)MC2_MALLOC(value->count * 4);
+
+        if (value->count == 0)
+            continue;
+
+        uint32_t values_index = 0;
+        do {
+            char *cur_value_str = value_str;
+            if (*value_str != '\0') {
+
+                do {
+                    if (*value_str == ' ')
+                        break;
+                } while (*value_str++ != '\0');
+
+                if (*value_str != '\0') {
+                    *value_str++ = '\0';
+                }
+            }
+            
+            value->args[values_index++] = MC2_STRDUP(cur_value_str);
+
+        } while (values_index < value->count);
+    }
+    
+    glo_8600F8.loc_611DC0();
+    return;
+}
+
+void sub_612F00() {
```

Comment:
Since this seems to be related to `unk_8600F8` (which the only known instance of the class AFAIK is `glo_8600F8`), this should probably be moved to that file eventually.